### ISSUE TITLE
[BUG] Error when creating a DataFrame from series of length 1

### ISSUE THREAD

**Issue by tebeka** [rapidsai/cudf#972]
**Describe the bug**

When initializing a cudf.DataFrame with series of size 1, we get a `TypeError`

**Steps/Code to reproduce bug**
Follow this guide http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

```python
import pandas as pd
from io import StringIO

data = '''
2019-02-15 15:34:05	4343893.00	volume	NASDAQ	AMZN	avg
2019-02-15 15:34:05	4343893.00	volume	NASDAQ	AMZN	max
2019-02-15 15:34:05	4343893.00	volume	NASDAQ	AMZN	last
2019-02-15 15:34:05	51.72	price	NYSE	ORCL	avg
2019-02-15 15:34:05	51.72	price	NYSE	ORCL	max
'''

columns = 'Date values metric_name exchange symbol Aggregate'.split()

io = StringIO(data)
df = pd.read_csv(io, sep='\t', names=columns, parse_dates=['Date'])
df.index = df.pop('Date')
for col in columns[2:]:
    df[col] = df[col].astype('category')

def from_df(df, size):
    data = {col: df[col][:size] for col in df.columns}
    return cudf.DataFrame(data)

from_df(df, 2)  # Will work
from_df(df, 1)  # Will raise "TypeError: data type not understood"
```

**Expected behavior**

Expect a DataFrame to be created even when the series size is 1.

**Environment details (please complete the following information):**
 - Environment location: [Bare-metal, Docker, Cloud(specify cloud provider)]
 - Method of cuDF install: [conda, Docker, or from source]
   - If method of install is [Docker], provide `docker pull` & `docker run` commands used
 - Please run and attach the output of the `cudf/print_env.sh` script to gather relevant environment details
 
install by conda, cudf 0.5.1

**Additional context**
None

**Comment by quasiben** [comment]
@tebeka thanks for the detailed bug report.  I just tried against the `branch-0.6` branch and all seems to work.  If possible can you build from source and try again ?

```
In [18]: from_df(df, 2).to_pandas()
Out[18]:
                        values metric_name exchange symbol Aggregate
Date
2019-02-15 15:34:05  4343893.0      volume   NASDAQ   AMZN       avg
2019-02-15 15:34:05  4343893.0      volume   NASDAQ   AMZN       max

In [19]: from_df(df, 1).to_pandas()
Out[19]:
                        values metric_name exchange symbol Aggregate
Date
2019-02-15 15:34:05  4343893.0      volume   NASDAQ   AMZN       avg
```

**Comment by kkraus14** [comment]
Fixed by #877
